5102-FastTable-Horizontal-Scrolling-not-implemented

danil-osipchuk opened the pull request:
A simple implementation of horizontal scrolling for FTTableMorph, to enable one has to set trialHSB:=true during initialization, otherwise current behavior is unchanged.
When enabled FTTableContainerMorph  allocates space to columns (#calculateColumnWidths) starting (#columnOrderOfWidthDistribution) from startColumnIndex ( instance var added with this commit, when set to 0 used as a flag to activate current/default behavior )  corresponding to the horizontal bar position triggered by  #adjustToHorizontalScrollBarValue: 
Only columns which manage to get space are being drawn  (#exposedColumnsRange: )


FTTableMorph newTrialHorizontalScrollBar                  <-- existing example
FTTableMorph>>#initialize                                                     <--  add trialHSB := true.  to enable it everywhere

danil-osipchuk committed:
A simple implementation of horizontal scrolling for FTTableMorph
FTTableContainerMorph starts drawing from the column index corresponding to the horizontal bar position

FTTableMorph newTrialHorizontalScrollBar                  <-- existing example
FTTableMorph>>#initialize                                                     <--  add trialHSB := true.  to enable it everywhere

estebanlm commented:
but... I would change `trialHSB` with `horizontalScrollBar` (since is not a trial anymore)

danil-osipchuk committed:
Removing trialHSB var, its role of a flag enabling  horizontal scrolling goes to horizontalScrollBar itself (var holding ScrollBar morph).
If it set at early initialization stage in #initializeScrollBars  and ScrollBar is being put there and used thereafter.
This is to gradually introduce the feature on per need basis.

danil-osipchuk commented:
trialHSB was used as a flag to enable initialization of a (existing) horizontalScrollBar with ScrollBar instance. I've nuked trialHSB and am using horizontalScrollBar itself for that role to have it disabled for now by default.
If we are to decide to enable it permanently all is needed is to remove  'horizontalScrollBar ifNotNil:' conditions in FTTableMorph

estebanlm commented:
errors not related
